src/uct/tcp/tcp_iface.c

```diff
@@ -694,17 +655,14 @@ static UCS_CLASS_INIT_FUNC(uct_tcp_iface_t, uct_md_h md, uct_worker_h worker,
             ucs_time_from_sec(UCT_TCP_EP_DEFAULT_KEEPALIVE_IDLE);
     }
 
-    status = uct_tcp_iface_set_port_range(self, config);
-    if (status != UCS_OK) {
-        goto err;
-    }
-
     if (self->config.tx_seg_size > self->config.rx_seg_size) {
         ucs_error("RX segment size (%zu) must be >= TX segment size (%zu)",
                   self->config.rx_seg_size, self->config.tx_seg_size);
-        return UCS_ERR_INVALID_PARAM;
+        status = UCS_ERR_INVALID_PARAM;
+        goto err;
     }
 
+    uct_tcp_iface_set_port_range(self, config);
```

Comment:
minor: may be expand uct_tcp_iface_set_port_range here? there is only 2 lines of code + debug now

Reply:
@hoopoepg fixed, and also removed the debug print, since uct_tcp_iface_listener_init already prints the selected server port number.